search/src/Components/upload/upload_stepper.js

```diff
@@ -299,6 +288,39 @@ class UploadStepper extends React.Component {
         }
     }
 
+    isNextEnabled() {
+        const step = stepsByType[this.props.upload_type][this.state.activeStep].type
+        const currentStageValid = (step) => {
+            switch (step) {
+                case "categories":
+                    return this.state.categories_saved
+                case "agcontext":
+                    return this.state.coco_form_validation['agcontexts']
+                case "metadata":
+                    return this.state.coco_form_validation['metadata']
+                case "images":
+                    return this.state.imageReady
+                default:
+                    return true
+            }
+        }
+        return currentStageValid(step) && this.state.error_message.length === 0
+    }
+
+    nextHandler() {
+        const step = stepsByType[this.props.upload_type][this.state.activeStep].type
+        switch (step) {
+            case "categories":
+                return this.handleUpdateCategories
+            case "agcontext":
+                return this.handleUploadAgcontexts
+            case "metadata":
+                return this.handleUploadMetadata
+            default:
+                return this.handleNext
+        }
+    }
+
```

Comment:
I would prefer if these step definitions were found together with `getStepContent`. So for example, we could rename `getStepContent` to `getStep` and it would return `{component: <Xyz />, handleNext: this.handleXyz}`

Reply:
I put isNextEnabled and nextHandler together, which makes more sense IMO. Do you also want to include the component in?

Reply:
> I put isNextEnabled and nextHandler together, which makes more sense IMO.

It doesn't make sense to me, because `isNextEnabled` is a property of the current state of the upload stepper (i.e. what data the user has put in), while `nextHandler` is a property of the step definition. Applying separation of concerns principles (encapsulation, polymorphism), the nextHandler belongs with the step component definition, not together with data-driven operational concerns